crates/bevy_ecs/src/hierarchy.rs

```diff
@@ -123,32 +123,32 @@ impl Deref for Parent {
     }
 }
 
-// TODO: We need to impl either FromWorld or Default so Parent can be registered as Reflect.
+// TODO: We need to impl either FromWorld or Default so ChildOf can be registered as Reflect.
 // This is because Reflect deserialize by creating an instance and apply a patch on top.
-// However Parent should only ever be set with a real user-defined entity.  Its worth looking into
+// However ChildOf should only ever be set with a real user-defined entity.  Its worth looking into
 // better ways to handle cases like this.
-impl FromWorld for Parent {
+impl FromWorld for ChildOf {
     #[inline(always)]
     fn from_world(_world: &mut World) -> Self {
-        Parent(Entity::PLACEHOLDER)
+        ChildOf(Entity::PLACEHOLDER)
     }
 }
 
 /// A [`RelationshipTarget`](crate::relationship::RelationshipTarget) collection component that is populated
-/// with entities that "target" this entity with the [`Parent`] [`Relationship`](crate::relationship::Relationship) component.
+/// with entities that "target" this entity with the [`ChildOf`] [`Relationship`](crate::relationship::Relationship) component.
 ///
-/// Together, these components form the "canonical parent-child hierarchy". See the [`Parent`] component for all full
+/// Together, these components form the "canonical parent-child hierarchy". See the [`ChildOf`] component for all full
 /// description of this relationship and instructions on how to use it.
 #[derive(Component, Default, VisitEntitiesMut, Debug, PartialEq, Eq)]
-#[relationship_target(relationship = Parent, despawn_descendants)]
+#[relationship_target(relationship = ChildOf, despawn_descendants)]
 #[cfg_attr(feature = "bevy_reflect", derive(bevy_reflect::Reflect))]
 #[cfg_attr(
     feature = "bevy_reflect",
     reflect(Component, MapEntities, VisitEntities, VisitEntitiesMut, FromWorld)
 )]
-pub struct Children(Vec<Entity>);
+pub struct ParentOf(Vec<Entity>);
```

Comment:
Alternatively I like `XTo` as well. And it may be possible to use that sort of naming convention to convey certain meaning such as ownership or directionality (or sources vs targets).